worlds/lingo/static_logic.py

```diff
@@ -0,0 +1,372 @@
+
+                        self.DOORS[door_obj.item_name] = door_obj
+                        self.DOORS_BY_ROOM[room_name][door_name] = door_obj
+
+                if "paintings" in room_data:
+                    self.PAINTINGS_BY_ROOM[room_name] = []
+
+                    for painting_data in room_data["paintings"]:
+                        painting_id = painting_data["id"]
+
+                        if "orientation" in painting_data:
+                            orientation = painting_data["orientation"]
+                        else:
+                            orientation = ""
+
+                        if "disable" in painting_data:
+                            disable_painting = painting_data["disable"]
+                        else:
+                            disable_painting = False
+
+                        if "required" in painting_data:
+                            required_painting = painting_data["required"]
+                            if required_painting:
+                                self.REQUIRED_PAINTING_ROOMS.append(room_name)
+                        else:
+                            required_painting = False
+
+                        if "move" in painting_data:
+                            move_painting = painting_data["move"]
+                        else:
+                            move_painting = False
+
+                        if "required_when_no_doors" in painting_data:
+                            rwnd = painting_data["required_when_no_doors"]
+                            if rwnd:
+                                self.REQUIRED_PAINTING_WHEN_NO_DOORS_ROOMS.append(room_name)
+                        else:
+                            rwnd = False
+
+                        if "exit_only" in painting_data:
+                            exit_only = painting_data["exit_only"]
+                        else:
+                            exit_only = False
+
+                        if "enter_only" in painting_data:
+                            enter_only = painting_data["enter_only"]
+                        else:
+                            enter_only = False
+
+                        required_door = None
+                        if "required_door" in painting_data:
+                            door = painting_data["required_door"]
+                            required_door = RoomAndDoor(
+                                door["room"] if "room" in door else room_name,
+                                door["door"]
+                            )
+
+                        painting_obj = Painting(painting_id, room_name, enter_only, exit_only, orientation,
+                                                required_painting, rwnd, required_door, disable_painting, move_painting)
+                        self.PAINTINGS[painting_id] = painting_obj
+                        self.PAINTINGS_BY_ROOM[room_name].append(painting_obj)
```

Comment:
Also, holy indent depth and function length.

I know you took inspiration from WitnessLogic.txt here and probably said "It should be in an actual data format like .yaml", but the reason I used .txt is so I can parse it line by line, object by object. For example: If a line is a "region line", I create it and then keep track of the "current_region". Then, for panels in that region, I reference "current_region". Because of this, I have a much lower indent depth. I also treat every "entity" the same, with optional arguments based on whether it is a panel, door, laser, etc. This means I don't have three seperate giant blocks for panels, doors and lasers.

I'm not saying that it makes my solution better, in fact it's probably way worse in terms of computation time, but I think if you're going to nest things like this, it'd be neat to seperate some of this stuff out into some functions.

like

```
    for room_name, room_data in config.items():
        create_room(room_name, room_data)

def create_room(room_data)
    ....
    if "panels" in room_data:
        create_panels(room_name, room_data["panels"])
    if "paintings" in room_data:
        create_paintings(room_name, room_data["paintings])
    #  Could even do this with setdefault({}) and get rid of the "if" statements
```

That sort of thing.

Reply:
Broke it out into functions, PTAL

Reply:
Looks good!